connector/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryProxyAndHttpConfig.java

```diff
@@ -0,0 +1,206 @@
+  private com.google.common.base.Optional<String> proxyUsername;
+  private com.google.common.base.Optional<String> proxyPassword;
+  private com.google.common.base.Optional<Integer> httpMaxRetry;
+  private com.google.common.base.Optional<Integer> httpConnectTimeout;
+  private com.google.common.base.Optional<Integer> httpReadTimeout;
+
+  @VisibleForTesting
+  SparkBigQueryProxyAndHttpConfig() {
+    // empty
+  }
+
+  @VisibleForTesting
+  public static SparkBigQueryProxyAndHttpConfig from(
+      Map<String, String> options,
+      ImmutableMap<String, String> globalOptions,
+      Configuration hadoopConfiguration)
+      throws IllegalArgumentException {
+    SparkBigQueryProxyAndHttpConfig config = new SparkBigQueryProxyAndHttpConfig();
+
+    com.google.common.base.Optional<String> proxyAddress =
+        getProperty(
+            options,
+            globalOptions,
+            hadoopConfiguration,
+            PROXY_ADDRESS_OPTION,
+            GCS_CONFIG_PROXY_ADDRESS_PROPERTY);
+    config.proxyUri = fromNullable(parseProxyAddress(proxyAddress.or("")));
+    config.proxyUsername =
+        getProperty(
+            options,
+            globalOptions,
+            hadoopConfiguration,
+            PROXY_USERNAME_OPTION,
+            GCS_CONFIG_PROXY_USERNAME_PROPERTY);
+    config.proxyPassword =
+        getProperty(
+            options,
+            globalOptions,
+            hadoopConfiguration,
+            PROXY_PASSWORD_OPTION,
+            GCS_CONFIG_PROXY_PASSWORD_PROPERTY);
+    checkProxyParamsValidity(config);
+
+    config.httpMaxRetry =
+        getProperty(
+                options,
+                globalOptions,
+                hadoopConfiguration,
+                HTTP_MAX_RETRY_OPTION,
+                GCS_CONFIG_HTTP_MAX_RETRY_PROPERTY)
+            .transform(Integer::valueOf);
+    config.httpConnectTimeout =
+        getProperty(
+                options,
+                globalOptions,
+                hadoopConfiguration,
+                HTTP_CONNECT_TIMEOUT_OPTION,
+                GCS_CONFIG_HTTP_CONNECT_TIMEOUT_PROPERTY)
+            .transform(Integer::valueOf);
+    config.httpReadTimeout =
+        getProperty(
+                options,
+                globalOptions,
+                hadoopConfiguration,
+                HTTP_READ_TIMEOUT_OPTION,
+                GCS_CONFIG_HTTP_READ_TIMEOUT_PROPERTY)
+            .transform(Integer::valueOf);
+    checkHttpParamsValidity(config);
+
+    return config;
+  }
+
+  private static void checkProxyParamsValidity(SparkBigQueryProxyAndHttpConfig config)
+      throws IllegalArgumentException {
+    if (!config.proxyUri.isPresent()
+        && (config.proxyUsername.isPresent() || config.proxyPassword.isPresent())) {
+      throw new IllegalArgumentException(
+          "if proxyAddress is null then proxyUsername and proxyPassword should be null");
+    }
+
+    if ((config.proxyUsername.isPresent() && !config.proxyPassword.isPresent())
```

Comment:
Should use `BigQueryProxyTransporterBuilder.checkProxyParamsValidity()` to avoid code duplication